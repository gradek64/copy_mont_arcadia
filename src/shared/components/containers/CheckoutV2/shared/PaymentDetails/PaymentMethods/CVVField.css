/** @define CVVField;weak */

.CVVField {
  &-row {
    flex-basis: 100%;
    display: flex;
    margin-top: $input-margin-vertical;
    align-items: center;
    margin:0;
  }

  &-cvv,
  &-cvvHelp {
    margin: 0 !important;
  }

  &-cvv {
    &--icon {
      input {
        background-image: url("/assets/common/images/icon-cvv.svg");
        background-repeat: no-repeat;
        background-position: right 15px center;
        background-size: 24px;
      }
    }

    &--warning {
      input {
        border: 1px solid #F79E1B !important;
      }
    }
  }

  &-cvvHelp {
    position: relative;
    top: 7px;
  }

  &-warningBox {
    border-top:  1px solid #F79E1B;
    border-bottom:  1px solid #F79E1B;
    background-color: #FFF9DE;
    padding: 17px 15px;
    margin: 30px 0;

    &--cardExpired {
      padding: 11px 50px 10px 15px;
      margin: 30px 0;
      display: flex;
      align-items: center;
    }
  }

  &-icon {
    width: 24px;
    height: 24px;
    vertical-align: middle;
    margin-right: 10px;
    &--expired {
      margin: 0 10px 0 0;
    }
  }

  &-icon {
    width: 24px;
    height: 24px;
    vertical-align: middle;
    margin-right: 10px;
  }

  &-link {
    white-space: nowrap;
    text-decoration: underline;
    margin-top: 50px;
    cursor: pointer;
    overflow: visible;
  }

  &-cvvInfo {
    margin: 10px 0 0;
  }
}

@media $min-tablet {
  .CVVField {
    &-row {
      flex-basis: 33.33%;
    }

    &-cvv {
      &--icon {
        input {
          width: 60%
        }
      }
    }
  }
}

@media $min-laptop {
  .CVVField {
    &-row {
      flex-basis: 50%;
    }

    &-cvv {
      &--icon {
        input {
          width: 45%
        }
      }
    }
  }
}
