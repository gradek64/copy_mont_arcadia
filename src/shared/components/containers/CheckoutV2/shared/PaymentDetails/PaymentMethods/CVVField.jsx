import PropTypes from 'prop-types'
import React, { Component } from 'react'
import classnames from 'classnames'

// components
import Input from '../../../../../common/FormComponents/Input/Input'
import Image from '../../../../../common/Image/Image'

export default class CVVField extends Component {
  static propTypes = {
    field: PropTypes.shape({
      value: PropTypes.string,
      isTouched: PropTypes.bool,
    }).isRequired,
    error: PropTypes.string,
    className: PropTypes.string,
    // functions
    setField: PropTypes.func.isRequired,
    touchedField: PropTypes.func.isRequired,
    messagePosition: PropTypes.oneOf(['default', 'bottom']),
    placeholder: PropTypes.string,
    label: PropTypes.string,
  }

  static defaultProps = {
    error: '',
    className: '',
    isMobile: false,
    messagePosition: 'default',
    placeholder: 'CVV*',
    label: 'CVV',
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  render() {
    const { l } = this.context
    const {
      field,
      error,
      className,
      setField,
      touchedField,
      messagePosition,
      placeholder,
      label,
      savedPaymentType,
      storedPaymentDetails,
      isStoredCardExpired,
    } = this.props

    const CVVFieldClass = classnames('CVVField', className)
    const CvvInputClass = classnames('CVVField-cvv', {
      'CVVField-cvv--icon': savedPaymentType,
      'CVVField-cvv--warning': !!error && savedPaymentType,
    })

    const cvvWarningBoxStyle = classnames('CVVField-warningBox', {
      'CVVField-warningBox--cardExpired': isStoredCardExpired,
    })

    const cvvIconStyle = classnames('CVVField-icon', {
      'CVVField-icon--expired': isStoredCardExpired,
    })

    const fieldHasValue = field.value && field.value.length > 0

    return (
      <div className={CVVFieldClass}>
        {savedPaymentType &&
          !fieldHasValue &&
          !isStoredCardExpired && (
            <div className={cvvWarningBoxStyle}>
              <Image
                className={cvvIconStyle}
                src={'/assets/common/images/icon-cvv.svg'}
                alt=""
              />
              {l`Enter your CVV number to continue`}
            </div>
          )}
        {isStoredCardExpired ? (
          <div className={cvvWarningBoxStyle}>
            <Image
              className={cvvIconStyle}
              src={'/assets/common/images/icon-expired-card.svg'}
              alt=""
            />
            {l`Card ending in ${storedPaymentDetails.cardNumberStar.replace(
              /\*/g,
              ''
            )} has expired. Try a different card or payment method`}
          </div>
        ) : (
          <div className="CVVField-row">
            <Input
              className={CvvInputClass}
              field={field}
              name="cvv"
              label={l(label)}
              type="tel"
              placeholder={l(placeholder)}
              setField={setField}
              touchedField={touchedField}
              messagePosition={messagePosition}
              errors={{ cvv: l(error) }}
              isRequired
            />
          </div>
        )}
      </div>
    )
  }
}
