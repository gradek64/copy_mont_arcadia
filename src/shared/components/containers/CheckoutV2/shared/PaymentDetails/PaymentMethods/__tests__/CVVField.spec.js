import testComponentHelper from 'test/unit/helpers/test-component'

import CVVField from '../CVVField'

describe('<CVVField/>', () => {
  const renderComponent = testComponentHelper(CVVField)

  const baseProps = {
    field: {
      value: '123',
      isTouched: true,
    },
    setField: () => {},
    touchedField: () => {},
  }

  describe('@renders', () => {
    it('should render default state correctly', () => {
      const component = renderComponent(baseProps)
      expect(component.getTree()).toMatchSnapshot()
    })

    it('should show cvv errors if error is set', () => {
      const { wrapper } = renderComponent({
        ...baseProps,
        error: '3 digits required',
        isStoredCardExpired: false,
        savedPaymentType: { type: 'CARD' },
      })

      const input = wrapper.find('Connect(Input)')
      expect(input.prop('errors')).toEqual({
        cvv: '3 digits required',
      })
    })

    it('should show expired card message when card is expired', () => {
      const { wrapper } = renderComponent({
        ...baseProps,
        error: '3 digits required',
        isStoredCardExpired: true,
        storedPaymentDetails: {
          cardNumberStar: '1111',
        },
      })
      expect(wrapper.find('.CVVField-warningBox--cardExpired')).toHaveLength(1)
    })

    it('should not show enter CVV message when the payment type is not saved', () => {
      const { wrapper } = renderComponent({
        ...baseProps,
        error: '3 digits required',
        savedPaymentType: false,
        isStoredCardExpired: false,
      })
      expect(wrapper.find('.CVVField-warningBox')).toHaveLength(0)
    })

    it('should localise CVV label, placeholder and errors', () => {
      const { wrapper } = testComponentHelper(CVVField, {
        context: {
          l: (str) =>
            ({
              CVV: 'bileag',
              'CVV*': 'segnaposto',
              busted: 'kaput',
            }[str]),
        },
      })({
        ...baseProps,
        error: 'busted',
      })
      const input = wrapper.find('Connect(Input)')
      expect(input.prop('label')).toBe('bileag')
      expect(input.prop('placeholder')).toBe('segnaposto')
      expect(input.prop('errors')).toEqual({
        cvv: 'kaput',
      })
    })

    it('should add provided `className` to the element', () => {
      const { wrapper } = renderComponent({
        ...baseProps,
        className: 'test-class-name',
      })
      expect(wrapper.find('.CVVField.test-class-name')).toHaveLength(1)
    })

    describe('returning customer', () => {
      it('should render default state correctly', () => {
        const component = renderComponent(baseProps)
        expect(component.getTree()).toMatchSnapshot()
      })
      it('should show enter CVV warning when card is not expired and no value has been input', () => {
        const { wrapper } = renderComponent({
          ...baseProps,
          field: {
            value: '',
            isTouched: false,
          },
          error: '3 digits required',
          savedPaymentType: true,
          isStoredCardExpired: false,
        })
        expect(wrapper.find('.CVVField-warningBox')).toHaveLength(1)
      })

      it('should not show warning box and should show errors when CVV has erroneous value', () => {
        const { wrapper } = renderComponent({
          ...baseProps,
          field: {
            value: 'abc',
            isTouched: true,
          },
          error: '3 digits required',
          savedPaymentType: { type: 'CARD' },
        })
        const input = wrapper.find('Connect(Input)')
        expect(wrapper.find('.CVVField-warningBox')).toHaveLength(0)
        expect(wrapper.find('Connect(Input)').prop('className')).toContain(
          'CVVField-cvv--warning'
        )
        expect(wrapper.find('Connect(Input)').prop('className')).toContain(
          'CVVField-cvv--icon'
        )

        expect(input.prop('errors')).toEqual({ cvv: '3 digits required' })
      })
    })
  })
})
