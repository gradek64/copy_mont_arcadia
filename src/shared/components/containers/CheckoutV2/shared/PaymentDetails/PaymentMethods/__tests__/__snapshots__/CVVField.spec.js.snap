// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CVVField/> @renders returning customer should render default state correctly 1`] = `
<div
  className="CVVField"
>
  <div
    className="CVVField-row"
  >
    <Connect(Input)
      className="CVVField-cvv"
      errors={
        Object {
          "cvv": "",
        }
      }
      field={
        Object {
          "isTouched": true,
          "value": "123",
        }
      }
      isRequired={true}
      label="CVV"
      messagePosition="default"
      name="cvv"
      placeholder="CVV*"
      setField={[Function]}
      touchedField={[Function]}
      type="tel"
    />
  </div>
</div>
`;

exports[`<CVVField/> @renders should render default state correctly 1`] = `
<div
  className="CVVField"
>
  <div
    className="CVVField-row"
  >
    <Connect(Input)
      className="CVVField-cvv"
      errors={
        Object {
          "cvv": "",
        }
      }
      field={
        Object {
          "isTouched": true,
          "value": "123",
        }
      }
      isRequired={true}
      label="CVV"
      messagePosition="default"
      name="cvv"
      placeholder="CVV*"
      setField={[Function]}
      touchedField={[Function]}
      type="tel"
    />
  </div>
</div>
`;
